src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -338,11 +339,14 @@ export function DiscoverMainRoute({
     stateContainer,
   ]);
 
+  const { solutionNavId } = customizationContext;
+  const { rootProfileLoading } = useRootProfile({ solutionNavId });
```

Comment:
Yeah that's a good point, although I think I'd like to avoid supporting profiles in `DiscoverContainer` anyway if possible. I could change `useRootProfile` to listen for `core.chrome.getActiveSolutionNavId$` instead, but what if we just pass an empty `ProfilesManager` for now?

Reply:
> although I think I'd like to avoid supporting profiles in DiscoverContainer anyway if possible.

Does this mean those apps embedding the DiscoverContainer (such as LogsExplorer) won't benefit from the resolution logic built-in in Discover with the new data source and document profile providers?
Isn't it the purpose of making Discover aware of the data so that the other solutions embedding it get the ad-hoc behaviours out of the box? 
Let me know if I misunderstood this and what is the intended support here please.

Reply:
I'd agree with @tonyghiani here. For any future addition to the Logs Explorer we'd all benefit if we could instead add it to a matching Discover context.

What would be the downside of having the Logs Explorer explicitly resolve the root context (always to an obs root context)?

Reply:
Thanks for the clarity and managing the communication with PM/Director Davis!

I have clear the vision for this and the future of One Discover, and your reasoning makes sense.
On the other hand, it's even true there are still many differences between Discover and the embedded version, which means we have to maintain the same features in different parts of the codebase if we don't rely on a single mechanism.

For instance, we did some work to bring the Logs overview into the flyout so that every log user could use it based on the resolved context. What feels counter-productive is having vanilla Discover as the only app taking decisions based on context (enabling the flyout overview tab in this case), while also have in parallel the embedded version manually enable the same features (logs explorer should keep manually allowing that flyout to tab until we have feature parity and Logs Explorer can be removed).

The above would apply to other features and implementation could quickly diverge in how things are done until we have a good level of feature parity and we keep a One Discover.

If is a minimal effort, I would strongly opt for a single source of truth to resolve specific customizations, which would be the profile resolution we have worked on so far.

Let me know your thoughts, I'm happy to talk more about this if needed (I missed the engineering sync yesterday, not sure how in depth this was already covered, I'll check the recording)